core/java/iesi-core/src/main/java/io/metadew/iesi/connection/database/bigquery/BigqueryDatabaseService.java

```diff
@@ -0,0 +1,256 @@
+            return new BigqueryDatabase(bigqueryDatabaseConnection, schemaName);
+        }
+
+        //Individual parameters have been provided
+        String host = DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, hostKey);
+        int port = Integer.parseInt(DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, portKey));
+        String project = DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, projectKey);
+        int authValue = Integer.parseInt(DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, authValueKey));
+
+        switch (authValue) {
+            case 0:
+                bigqueryDatabaseConnection = new BigqueryDatabaseConnection(
+                        getConnectionUrlServiceAccount(
+                                host,
+                                port,
+                                project,
+                                authValue,
+                                DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, serviceAccountKey),
+                                DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, keyPathKey)));
+                return new BigqueryDatabase(bigqueryDatabaseConnection, schemaName);
+            case 1:
+                bigqueryDatabaseConnection = new BigqueryDatabaseConnection(
+                        getConnectionUrlUserAccount(
+                                host,
+                                port,
+                                project,
+                                authValue));
+                return new BigqueryDatabase(bigqueryDatabaseConnection, schemaName);
+            case 2:
+                bigqueryDatabaseConnection = new BigqueryDatabaseConnection(
+                        getConnectionUrlWithTokens (
+                                host,
+                                port,
+                                project,
+                                authValue,
+                                DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, accessTokenKey),
+                                DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, refreshTokenKey),
+                                DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, clientIdKey),
+                                DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, clientSecretKey)));
+                return new BigqueryDatabase(bigqueryDatabaseConnection, schemaName);
+            case 3:
+                bigqueryDatabaseConnection = new BigqueryDatabaseConnection(
+                        getConnectionUrlDefaultCredentials(
+                                host,
+                                port,
+                                project,
+                                authValue));
+                return new BigqueryDatabase(bigqueryDatabaseConnection, schemaName);
```

Comment:
Create 4 subclasses of BigqueryDatabaseConnection for each auth type. Constructor of each subclass only receives the necessary arguments needed for the specific auth type. See 'io.metadew.iesi.connection.database.oracle.OracleDatabaseService' and 'io.metadew.iesi.connection.database.oracle.ServiceNameOracleDatabaseConnection' for example

Reply:
Resolved in next commit